src/renderers/shared/fiber/ReactFiberPendingWork.js

```diff
@@ -56,71 +86,57 @@ exports.findNextUnitOfWorkAtPriority = function(currentRoot : Fiber, priorityLev
       // because it is the highest priority for the whole subtree.
       // TODO: Coroutines can have work in their stateNode which is another
       // type of child that needs to be searched for work.
-      if (current.childInProgress) {
-        let workInProgress = current.childInProgress;
-        while (workInProgress) {
-          workInProgress.return = current.alternate;
-          workInProgress = workInProgress.sibling;
+      if (workInProgress.childInProgress) {
+        let child = workInProgress.childInProgress;
+        while (child) {
+          child.return = workInProgress;
+          child = child.sibling;
         }
-        workInProgress = current.childInProgress;
-        while (workInProgress) {
-          // Don't bother drilling further down this tree if there is no child.
-          if (workInProgress.pendingWorkPriority !== NoWork &&
-              workInProgress.pendingWorkPriority <= priorityLevel &&
-              workInProgress.pendingProps !== null) {
-            return workInProgress;
+        child = workInProgress.childInProgress;
+        while (child) {
+          // Don't bother drilling further down this tree if there is no child
+          // with more content.
+          // TODO: Shouldn't this still drill down even though the first
+          // shallow level doesn't have anything pending on it.
```

Comment:
This todo is correct, I think. Once you split the priority fields, there could be a matching update further down the tree. Now that this function operates on the work in progress tree, could you do a recursive call?